### PR TITLE
Adds 1D geostrophic adjustment example

### PR DESCRIPTION
**Pull request by apaloczy**
I've written this simple 1D geostrophic adjustment problem in the process of learning to use FourierFlows.

### CODE REVIEW

**Comment by glwagner** [comment]
@apaloczy this looks great, and it runs on my laptop! 

My small request is: can you write a few sentences as comments at the top of the file about what equations this problem solves? A few comments in the problem set-up may also be nice so users understand what they are looking at. 

I also made some completely trivial comments about spaces after commas.

File: examples/OneDShallowWaterRossbyAdjustment.jl [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  ldiv!(v.ux, g.rfftplan, -im*g.kr.*v.uh)
+  ldiv!(v.vx, g.rfftplan, -im*g.kr.*v.vh)
+  mul!(v.uuxh, g.rfftplan, v.u.*v.ux)
+  mul!(v.uvxh, g.rfftplan, v.u.*v.vx)
+  rhsu = p.f*v.vh + p.g*im*g.kr.*v.etah - v.uuxh
+  rhsv = - p.f*v.uh - v.uvxh
+  rhseta = p.h*im*g.kr.*v.uh
+  N[:,1] .= rhsu
+  N[:,2] .= rhsv
+  N[:,3] .= rhseta
```
Comment (glwagner) [inline examples/OneDShallowWaterRossbyAdjustment.jl]: spaces after commas!

File: examples/OneDShallowWaterRossbyAdjustment.jl [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      f  :: T         # Coriolis parameter
+end
+Params(ν, nν, g, h, f) = Params(ν, nν, g, h, f)
+
+function Equation(p, g)
+  LC = zeros(Float64, g.nkr, 3)
+  D = - p.ν*g.kr.^p.nν
+  LC[:,1] .= D # u
+  LC[:,2] .= D # v
+  LC[:,3] .= D # eta
```
Comment (glwagner) [inline examples/OneDShallowWaterRossbyAdjustment.jl]: spaces after commas!

File: examples/OneDShallowWaterRossbyAdjustment.jl [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+end
+
+function set_uveta!(u0, v0, eta0, prob)
+    g, sol, p, v = prob.grid, prob.sol, prob.params, prob.vars
+    mul!(v.uh, g.rfftplan, u0)
+    mul!(v.vh, g.rfftplan, v0)
+    mul!(v.etah, g.rfftplan, eta0)
+    sol[:,1] = rfft(u0)
+    sol[:,2] = rfft(v0)
+    sol[:,3] = rfft(eta0)
```
Comment (glwagner) [inline examples/OneDShallowWaterRossbyAdjustment.jl]: spaces after commas!

**Comment by navidcy** [comment]
I agree with @glwagner about some comments on what equations this example is solving.
Also, I feel that it would more appropriately fit in [GeophysicalFlows.jl](https://github.com/FourierFlows/GeophysicalFlows.jl) repo.

**Comment by apaloczy** [comment]
I see, thanks for the feedback! I will close this pr and open one on GeophysicalFlows.jl with a new version of this example soon.

**Comment by glwagner** [comment]
Hey @apaloczy, just wanted to add that I do think this is an appropriate example for `FourierFlows`. It's short and sweet, and shows how to define a "Problem" module and also how to run it. And it's well-written.

To contribute this code to `GeophysicalFlows', it should probably be be split into two: Lines 1-88 would be partitioned into a module in `GeophysicalFlows/src` (and included at the top-level in `GeophysicalFlows.jl`, while lines 90-end would be siphoned into a example for `GeophysicalFlows/examples`. Eventually, as a full-fledged model in `GeophysicalFlows`, we would have to add documentation to make it useable for users.

Either type of contribution is 100% fine with me, of course! Thanks again for the code.

**Comment by navidcy** [comment]
@apaloczy, I think @glwagner is right. Perhaps my comment above was a bit rushed. The example would fit nicely here give some extra comments at the top of the .jl file that mention the exact equation that is being solved.

**Comment by apaloczy** [comment]
I see. I agree with @glwagner'a idea of making the problem definition part of the code a module of GeophysicalFlows.jl in the future, for shallow water models. I'll open a new pr here on FourierFlows with the commented code in a bit.

**Pull request closed without merging**